runtime/vm/classsupport.c

```diff
@@ -542,18 +542,17 @@ callFindLocallyDefinedClass(J9VMThread* vmThread, J9Module *j9module, U_8* class
 	J9ROMClass* returnVal = NULL;
 	IDATA classFound = 0;
 	IDATA* returnPointer = (IDATA*) &returnVal;
+	BOOLEAN isBootLoader = classLoader == vmThread->javaVM->systemClassLoader;
 
 	/* localBuffer should not be NULL */
 	Assert_VM_true(NULL != localBuffer);
 
 	if (NULL != dynamicLoadBuffers) {
-		 J9ClassPathEntry* classPathEntries = NULL;
-		 if (classLoader == vmThread->javaVM->systemClassLoader) {
-			 classPathEntries = classLoader->classPathEntries;
-		 }
-		 TRIGGER_J9HOOK_VM_FIND_LOCALLY_DEFINED_CLASS(vmThread->javaVM->hookInterface, vmThread, classLoader, j9module, (char*)className, classNameLength,
-						classPathEntries, classLoader->classPathEntryCount, -1, NULL, 0, 0,
+		if (isBootLoader) {
```

Comment:
This seems wrong. If we can get there for non-boostrap it should trigger the hook.

Reply:
Can we get here for non-bootstrap class loader ?

It is case 1 of https://github.com/eclipse-openj9/openj9/pull/13483#issuecomment-926201647, I thought we won't get here from a non-bootstrap loader.

Reply:
If that is the case then the checks for isBootLoader are redundant. I'd prefer an assert checking for the bootstrap loader.

Reply:
It is possible this can be non-bootstrap class loader reaching here called from `Fast_java_lang_VMAccess_findClassOrNull()` with `J9_FINDCLASS_FLAG_USE_LOADER_CP_ENTRIES.` 

From the flag name `J9_FINDCLASS_FLAG_USE_LOADER_CP_ENTRIES`, looks like it wants to find the class in the loader CPE. however, non-bootstrap class loader has no CPE when SCC is off. When SCC on, the CPE count is 0. The following call `findLocallyDefinedClassFunction()` will never return class from non-bootstrap class loader CPE. I guess the SCC hook should have the same behaviour as `findLocallyDefinedClassFunction()` which loads from file system. 

Seems this is more than what this PR should cover. I have recovered the original code.